discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -58,11 +60,13 @@ final class ServiceDiscovery(implicit system: ExtendedActorSystem) extends Exten
       dynamic
         .createInstanceFor[SimpleServiceDiscovery](clazzName, (classOf[ExtendedActorSystem] → system) :: Nil)
         .recoverWith {
-          case _ ⇒
+          case e: Throwable ⇒
```

Comment:
I think we should probably only catch things related to the constructor not being available e.g. NoSuchMethod and then NonFatail(e) for the the other exceptions. Up to you if you want to do this now, happy to merge as is as this is definitely an improvement

Reply:
Sounds good :+1: